examples/platform/silabs/SiWx917/SiWx917/rsi_if.c

```diff
@@ -195,9 +200,10 @@ static void wfx_rsi_join_cb(uint16_t status, const uint8_t * buf, const uint16_t
  *********************************************************************/
 static void wfx_rsi_join_fail_cb(uint16_t status, uint8_t * buf, uint32_t len)
 {
-    WFX_RSI_LOG("%s: error: failed status: %02x on try %d", __func__, status, wfx_rsi.join_retries);
+    WFX_RSI_LOG("%s: error: failed status: %02x", __func__, status);
     wfx_rsi.join_retries += 1;
-    wfx_rsi.dev_state &= ~WFX_RSI_ST_STA_CONNECTING;
+    wfx_rsi.dev_state &= ~(WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED);
+    is_wifi_disconnection_event = true;
```

Comment:
once set to true, is there anything setting it back to false? Or is there no need of such a thing? If no need, we should probably document it here or above when the variable is declared.

Reply:
After successful connection clearing **is_wifi_disconnection_event**.